sql/pipelines/src/test/scala/org/apache/spark/sql/pipelines/utils/PipelineTest.scala

```diff
@@ -0,0 +1,467 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.pipelines.utils
+
+import java.io.{BufferedReader, FileNotFoundException, InputStreamReader}
+import java.nio.file.Files
+
+import scala.collection.mutable.ArrayBuffer
+import scala.util.{Failure, Try}
+import scala.util.control.NonFatal
+
+import org.scalactic.source
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach, Tag}
+import org.scalatest.matchers.should.Matchers
+
+import org.apache.spark.{SparkConf, SparkFunSuite}
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.{Column, QueryTest, Row, TypedColumn}
+import org.apache.spark.sql.SparkSession.{clearActiveSession, setActiveSession}
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.classic.{DataFrame, Dataset, SparkSession, SQLContext}
+import org.apache.spark.sql.execution._
+import org.apache.spark.sql.pipelines.utils.PipelineTest.{cleanupMetastore, createTempDir}
+
+abstract class PipelineTest
+    extends SparkFunSuite
+    with BeforeAndAfterAll
+    with BeforeAndAfterEach
+    with Matchers
+    with SparkErrorTestMixin
+    with TargetCatalogAndDatabaseMixin
+    with Logging {
+
+  final protected val storageRoot = createTempDir()
+
+  var spark: SparkSession = createAndInitializeSpark()
+  val originalSpark: SparkSession = spark.cloneSession()
+
+  implicit def sqlContext: SQLContext = spark.sqlContext
+  def sql(text: String): DataFrame = spark.sql(text)
+
+  /**
+   * Spark confs for [[originalSpark]]. Spark confs set here will be the default spark confs for
+   * all spark sessions created in tests.
+   */
+  protected def sparkConf: SparkConf = {
+    new SparkConf()
+      .set("spark.sql.shuffle.partitions", "2")
+      .set("spark.sql.session.timeZone", "UTC")
+  }
+
+  /** Returns the dataset name in the event log. */
+  protected def eventLogName(
+      name: String,
+      catalog: Option[String] = catalogInPipelineSpec,
+      database: Option[String] = databaseInPipelineSpec,
+      isView: Boolean = false
+  ): String = {
+    fullyQualifiedIdentifier(name, catalog, database, isView).unquotedString
+  }
+
+  /** Returns the fully qualified identifier. */
+  protected def fullyQualifiedIdentifier(
+      name: String,
+      catalog: Option[String] = catalogInPipelineSpec,
+      database: Option[String] = databaseInPipelineSpec,
+      isView: Boolean = false
+  ): TableIdentifier = {
+    if (isView) {
+      TableIdentifier(name)
+    } else {
+      TableIdentifier(
+        catalog = catalog,
+        database = database,
+        table = name
+      )
+    }
+  }
+
+  /**
+   * Runs the given function with the given spark conf, and resets the conf after the function
+   * completes.
+   */
+  def withSparkConfs[T](confs: Map[String, String])(f: => T): T = {
```

Comment:
shall we reuse the existing withSQLConf?

Reply:
makes sense. removed this func.